package/MDAnalysis/coordinates/__init__.py

```diff
@@ -532,6 +532,11 @@ class can choose an appropriate reader automatically.
      :class:`MDAnalysis.coordinates.base.ProtoReader.__iter__` (which is always
      implemented) and other slices raise :exc:`TypeError`.
 
+ ``parse_n_atoms(filename)``
+     Provide the number of atoms in the trajectory file, allowing the Reader
+     to be used to provide an extremely minimal Topology.
+     Must be implemented as either a staticmethod or a classmethod.
```

Comment:
what is the benefit of this being a classmethod?

Reply:
most Reader init's do a lot of work (ie load the first frame), so I wanted to avoid that, staticmethod/classmethod allows that.  Some classes use parts of the class to read a file (eg `XDRReader._file`) so need to use a classmethod not just a staticmethod